tests/utilities/test_apply_func.py

```diff
@@ -115,24 +150,54 @@ class ModelExample:
         reduced["h"].segment_ids, expected_result["h"].segment_ids
     ), "Reduction of a dataclass did not yield the desired result"
 
-    assert dataclasses.is_dataclass(reduced["i"]) and not isinstance(
-        reduced["i"], type
-    ), "Reduction of a dataclass should result in a dataclass"
-    assert dataclasses.is_dataclass(reduced["i"].feature) and not isinstance(
-        reduced["i"].feature, type
-    ), "Reduction of a nested dataclass should result in a nested dataclass"
-    assert (
-        reduced["i"].example_ids == expected_result["i"].example_ids
-    ), "Reduction of a nested dataclass did not yield the desired result"
+    def _assert_nested_dataclass_reduction(actual: ModelExample, expected: ModelExample):
+
+        assert dataclasses.is_dataclass(actual) and not isinstance(
+            actual, type
+        ), "Reduction of a dataclass should result in a dataclass"
+        assert dataclasses.is_dataclass(actual.feature) and not isinstance(
+            actual.feature, type
+        ), "Reduction of a nested dataclass should result in a nested dataclass"
+        assert (
+            actual.example_ids == expected.example_ids
+        ), "Reduction of a nested dataclass did not yield the desired result"
+        assert torch.allclose(
```

Comment:
these should equal, not just "close" :)

Reply:
Since these tensors were float like all of the rest, I interpreted this comment to mean that all applications of `allclose` should be replaced with `equal` and have done so throughout the test.

Reply:
I think your additions to the test should assert equality. There is no reason to assume the value is changed by any operations in `apply_to_collection`.